internal/config/client.go

```diff
@@ -40,6 +44,37 @@ const (
 	expectContinueTimeout = 1 * time.Second
 )
 
+type AuthMethod int
+
+const (
+	ServiceAccount AuthMethod = iota
+	Digest
+	Unknown
+)
+
+// CredentialProvider interface for types that can provide MongoDB Atlas credentials
+type CredentialProvider interface {
+	GetPublicKey() string
+	GetPrivateKey() string
+	GetClientID() string
+	GetClientSecret() string
+}
+
+// IsDigestAuth checks if public/private key credentials are present
+func IsDigestAuth(publicKey, privateKey string) bool {
+	return publicKey != "" && privateKey != ""
+}
+
+// IsServiceAccountAuth checks if client ID/secret credentials are present
+func IsServiceAccountAuth(clientID, clientSecret string) bool {
+	return clientID != "" && clientSecret != ""
+}
+
+// HasValidAuthCredentials checks if any valid authentication method is provided
+func HasValidAuthCredentials(publicKey, privateKey, clientID, clientSecret string) bool {
+	return IsDigestAuth(publicKey, privateKey) || IsServiceAccountAuth(clientID, clientSecret)
+}
```

Comment:
```suggestion
func IsDigestAuth(cp CredentialProvider) bool {
	return cp.GetPublicKey != "" && cp.GetPrivateKey != ""
}

// IsServiceAccountAuth checks if client ID/secret credentials are present
func IsServiceAccountAuth(cp CredentialProvider) bool {
	return cp.GetClientID != "" && cp.GetClientSecret != ""
}

// HasValidAuthCredentials checks if any valid authentication method is provided
func HasValidAuthCredentials(cp CredentialProvider) bool {
	return IsDigestAuth(cp) || IsServiceAccountAuth(cp)
}
```

interface can be used so need to pass explicitly the variables, similar to ResolveAuthMethod param

Reply:
done in [106af6b](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3700/commits/106af6bd4e9416b9fa787c48f30b21cebc25da76)